plugins/modules/remote_management/redfish/ilo_redfish_config.py

```diff
@@ -0,0 +1,169 @@
+version_added: 4.1.0
+description:
+    - "Provides an interface to manage configuration attributes."
+options:
+  category:
+    required: true
+    type: str
+    description:
+      - Command category to execute on iLO
+    choices: ['Manager', 'Sessions']
+  command:
+    required: true
+    description:
+      - List of commands to execute on iLO
+    type: list
+    elements: str
+  baseuri:
+    required: true
+    description:
+      - Base URI of iLO
+    type: str
+  username:
+    description:
+      - User for authentication with iLO
+    type: str
+  password:
+    description:
+      - Password for authentication with iLO
+    type: str
+  auth_token:
+    description:
+      - Security token for authentication with OOB controller
+    type: str
+  timeout:
+    description:
+      - Timeout in seconds for URL requests to iLO controller
+    default: 10
+    type: int
+  attribute_name:
+    required: false
+    description:
+      - Name of the attribute
+    type: str
+  attribute_value:
+    required: false
+    description:
+      - Value of the attribute
+    type: str
+requirements:
+    - "python >= 3.8"
+    - "ansible >= 3.2"
+author:
+    - "Bhavya B (@bhavya06)"
+'''
+
+EXAMPLES = '''
+  - name: Disable WINS Registration
+    community.general.ilo_redfish_config:
+      category: Manager
+      command: SetWINSReg
+      baseuri: 15.X.X.X
+      username: Admin
+      password: Testpass123
+      attribute_name: WINSRegistration
+
+  - name: Set Time Zone
+    community.general.ilo_redfish_config:
+      category: Manager
+      command: SetTimeZone
+      baseuri: 15.X.X.X
+      username: Admin
+      password: Testpass123
+      attribute_name: TimeZone
+      attribute_value: Chennai
+'''
+
+CATEGORY_COMMANDS_ALL = {
+    "Manager": ["SetTimeZone", "SetDNSserver", "SetDomainName", "SetNTPServers", "SetWINSReg"],
+    "Sessions": ["GetiLOSessions"]
+}
+
+from ansible_collections.community.general.plugins.module_utils.ilo_redfish_utils import iLORedfishUtils
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+
+
+def main():
+    result = {}
+    module = AnsibleModule(
+        argument_spec=dict(
+            category=dict(required=True, choices=list(
+                CATEGORY_COMMANDS_ALL.keys())),
+            command=dict(required=True, type='list', elements='str'),
+            baseuri=dict(required=True),
+            username=dict(),
+            password=dict(no_log=True),
+            attribute_name=dict(),
+            attribute_value=dict(),
+            auth_token=dict(no_log=True),
+            timeout=dict(type='int', default=10)
+        ),
+        supports_check_mode=False
+    )
+
+    category = module.params['category']
+    command_list = module.params['command']
+
+    session_id = None
+    creds = {"user": module.params['username'],
+             "pswd": module.params['password'],
+             "token": module.params['auth_token']}
```

Comment:
Actually, I checked the code of the base `RedFishUtils` class and it seems that `token` is mutually exclusive with both `user` and `pwsd` and that these two are required together. Ansible does have a neat way to express that, check on other modules in this very repository and you will find many examples.

Reply:
In fact the pre-existing RedFish modules (`redfish_config` for example) these constraints are present. Since that part of the code is shared, it only makes sense to use the same construct.

Reply:
constraints are added